File: FruitWholesale.Client/src/app/features/daily-expense/daily-expense.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { PaginatedList, PaginationRequest } from '../../core/models/common.model';
import { DailyExpense, SaveDailyExpense } from '../../core/models/transactions.model';
import { toHttpParams } from '../../core/utils/http-params.util';

@Injectable({ providedIn: 'root' })
export class DailyExpenseService {
  private readonly baseUrl = `${environment.apiUrl}/dailyexpense`;

  constructor(private readonly http: HttpClient) {}

  getPaged(request: PaginationRequest, expenseCategoryId?: number | null, fromDate?: string | null, toDate?: string | null) {
    return this.http.get<PaginatedList<DailyExpense>>(this.baseUrl, {
      params: toHttpParams({ ...request, expenseCategoryId, fromDate, toDate })
    });
  }

  create(dto: SaveDailyExpense): Observable<DailyExpense> {
    return this.http.post<DailyExpense>(this.baseUrl, dto);
  }

  update(id: number, dto: SaveDailyExpense): Observable<DailyExpense> {
    return this.http.put<DailyExpense>(`${this.baseUrl}/${id}`, dto);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}

File: FruitWholesale.Mobile/lib/features/daily_expense/daily_expense_service.dart
import '../../core/api/api_client.dart';
import '../../core/models/paginated_list.dart';
import 'daily_expense_models.dart';

class DailyExpenseService {
  final ApiClient _api;

  DailyExpenseService(this._api);

  Future<PaginatedList<DailyExpense>> getPaged({int pageNumber = 1, int pageSize = 20, String? fromDate, String? toDate}) async {
    final json = await _api.get('/dailyexpense', query: {
      'pageNumber': pageNumber,
      'pageSize': pageSize,
      if (fromDate != null) 'fromDate': fromDate,
      if (toDate != null) 'toDate': toDate,
    });
    return PaginatedList.fromJson(json as Map<String, dynamic>, DailyExpense.fromJson);
  }

  Future<DailyExpense> getById(int id) async {
    final json = await _api.get('/dailyexpense/$id');
    return DailyExpense.fromJson(json as Map<String, dynamic>);
  }

  Future<DailyExpense> create(DailyExpense expense) async {
    final json = await _api.post('/dailyexpense', body: expense.toSaveJson());
    return DailyExpense.fromJson(json as Map<String, dynamic>);
  }

  Future<DailyExpense> update(int id, DailyExpense expense) async {
    final json = await _api.put('/dailyexpense/$id', body: expense.toSaveJson());
    return DailyExpense.fromJson(json as Map<String, dynamic>);
  }
}
